newsroom/users/users.py

```diff
@@ -149,3 +149,8 @@ def password_match(self, password, hashed_password):
 
     def on_deleted(self, doc):
         app.cache.delete(str(doc.get('_id')))
+
+    def on_delete(self, doc):
+        if doc.get('_id') == get_user_id():
+            raise Exception
+        app.cache.delete(str(doc.get('_id')))
```

Comment:
This should not be needed, as we already have this in the `on_deleted` function, which is called **after** the User is deleted